File: showcase/test-samples.js
// JavaScript sample code
function greet(name) {
    const message = `Hello, ${name}!`;
    console.log(message);
    return message;
}

class Person {
    constructor(name, age) {
        this.name = name;
        this.age = age;
    }

    sayHello() {
        return `Hi, I'm ${this.name}`;
    }
}

const person = new Person("Alice", 30);
greet(person.name);

// Keywords: function, const, class, constructor, this, return, new
// Strings: "Hello, ${name}!", "Hi, I'm ${this.name}", "Alice"
// Comments: none in this sample
// Functions: greet, sayHello
// Variables: message, person, name, age
// Numbers: 30
// Operators: =, ., ${}, !
// Classes: Person
// Constants: none explicit

File: showcase/test-samples.rs
// Rust sample code
fn greet(name: &str) -> String {
    let message = format!("Hello, {}!", name);
    println!("{}", message);
    message
}

struct Person {
    name: String,
    age: u32,
}

impl Person {
    fn new(name: String, age: u32) -> Person {
        Person { name, age }
    }

    fn say_hello(&self) -> String {
        format!("Hi, I'm {}", self.name)
    }
}

let person = Person::new("Alice".to_string(), 30);
greet(&person.name);

// Keywords: fn, let, struct, impl, return, println
// Strings: "Hello, {}!", "Hi, I'm {}", "Alice"
// Comments: // Rust sample code, // Keywords: ...
// Functions: greet, say_hello, new
// Variables: message, person, name, age
// Numbers: 30
// Operators: =, ., &, ::, {}, !
// Classes: Person (struct)
// Constants: none